common/lib/xmodule/xmodule/js/src/video/01_initialize.js

```diff
@@ -532,12 +532,9 @@ function (VideoPlayer, VideoStorage, i18n) {
 
             this.youtubeXhr
                 .always(function (json, status) {
-                    var err = $.isPlainObject(json.error) ||
-                                (
-                                    status !== 'success' &&
-                                    status !== 'notmodified'
-                                );
-                    if (err) {
+                    // It will work for both if statusCode is 200 or 410.
+                    var didSucceed = (json.error && json.error.code === 410) || status === 'success';
```

Comment:
The api will only every return a 410 now, so there's no reason to filter on it. Let's just check for `$.isPlainObject(json.error) || status === 'success'`